gdal_subsetter/transform.py

```diff
@@ -1019,107 +1012,92 @@ def get_bbox_lonlat(self, filename):
         ds = None
         return bbox
 
-    def checkstackable(self, filelist):
-        '''
-        input:  list of the geotiff file names.
-        return: True/False to indicate that if the files in the filelist can be stacked.
-        only files with the same projection and geotransform can be stacked.
-        '''
-        projlst=[]
-        gtlst=[]
-        dtypelst=[]
-        xsizelst=[]
-        ysizelst=[]
-        formatlst=[]
-        stackable = False
-
-        for item in filelist:
-            ds = gdal.Open(item)
-            projlst.append(ds.GetProjection())
-            gtlst.append("{gt}".format( gt = ds.GetGeoTransform() ) )
-            dtypelst.append("{dtype}".format( dtype = ds.GetRasterBand(1).DataType))
-            xsizelst.append("{xsize}".format( xsize = ds.RasterXSize))
-            ysizelst.append("{ysize}".format( ysize = ds.RasterYSize))
-            formatlst.append(os.path.splitext(item)[1])
-
-        result_proj = False
-        if len(projlst) > 0 :
-            result_proj = all(elem == projlst[0] for elem in projlst)
-
-        result_gt = False
-        if len(gtlst) > 0 :
-            result_gt = all(elem == gtlst[0] for elem in gtlst)
-
-        result_dtype = False
-        if len(dtypelst) > 0 :
-            result_dtype = all(elem == dtypelst[0] for elem in dtypelst)
-
-        result_xsize = False
-        if len(xsizelst) > 0 :
-            result_xsize = all(elem == xsizelst[0] for elem in xsizelst)
-
-        result_ysize = False
-        if len(ysizelst) > 0 :
-            result_ysize = all(elem == ysizelst[0] for elem in ysizelst)
-
-        result_format = True
-        if '.png' in formatlst:
-            result_format = False
-
-        if result_proj and result_gt and result_dtype and result_xsize and result_ysize and result_format:
-            stackable = True
-
-        return stackable
+    def is_stackable(self, files_to_stack: str) -> bool:
```

Comment:
This is one of the two main places where I've tweaked things more significantly. The behaviour should be the same as the older version, though (and now we have accompanying unit tests to ensure that functionality persists through future iterations).

Part of me was tempted to extract this into a new module, where it doesn't actually make use of any of the other class attributes or methods, but I thought I'd start with just a quick refactor in place for now.